examples/1d/elixir_advection_basic.jl

```diff
@@ -10,17 +10,26 @@ equations = LinearScalarAdvectionEquation1D(advectionvelocity)
 
 initial_condition = initial_condition_convergence_test
 
+# you can either use a single function to impose the BCs weakly in all
+# 1*ndims == 2 directions or you can pass a tuple containing BCs for
+# each direction
+# Note: "boundary_condition_periodic" indicates that it is a periodic boundary and can be omitted on
+#       fully periodic domains. Here, however, it is included to allow easy override during testing
+boundary_conditions = boundary_condition_periodic
+
 surface_flux = flux_lax_friedrichs
 solver = DGSEM(3, surface_flux)
 
 coordinates_min = (-1,)
 coordinates_max = ( 1,)
 mesh = TreeMesh(coordinates_min, coordinates_max,
                 initial_refinement_level=4,
-                n_cells_max=30_000)
+                n_cells_max=30_000,
+                periodicity=true)
 
 
-semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
+                                    boundary_conditions=boundary_conditions)
```

Comment:
Why did you add the `boundary_conditions` here? We didn't do that for other basic examples...

Reply:
If you just want to be able to test the `*_linear_x` stuff, you could also create a new elixir. That would reduce the complexity of the most simple example, I think.

Reply:
Yes, but on the other hand this increases the complexity of verifying Taal against Taam... If it's ok by you, I'd just add
```julia
# TODO Taal: create separate `*_linear_x` elixir to keep this one simple
```
to the tests file so that we can decide at a later point

Reply:
Okay

Reply:
Done in 7e46b5e.